entities/src/main/java/com/flipkart/varadhi/entities/MessageWithOffset.java

```diff
@@ -0,0 +1,9 @@
+package com.flipkart.varadhi.entities;
+
+public interface MessageWithOffset<O extends Offset> extends Message {
```

Comment:
Does it make sense to merge it with PolledMessage, when it is needed independently meaning if there is Offset associated with message it should have other properties like topic/partition etc ?

Reply:
Agree; we don't need a separate message with offset.
PolledMessage acts as the Message contract from the consumer side. It has the info on partitions and offsets.